client/telemetry/TelemetryEventClient.go

```diff
@@ -283,3 +284,53 @@ func (impl *TelemetryEventClientImpl) HeartbeatEventForTelemetry() {
 		Properties: prop,
 	})
 }
+
+
+func (impl *TelemetryEventClientImpl) GetUCID() (*PosthogData, error) {
+	clusterBean, err := impl.clusterService.FindOne(cluster.ClusterName)
+	if err != nil {
+		impl.logger.Errorw("exception while getting unique client id", "error", err)
+		return nil, err
+	}
+	cfg, err := impl.clusterService.GetClusterConfig(clusterBean)
+	if err != nil {
+		impl.logger.Errorw("exception while getting unique client id", "error", err)
+		return nil, err
+	}
+	client, err := impl.K8sUtil.GetClient(cfg)
+	if err != nil {
+		impl.logger.Errorw("exception while getting unique client id", "error", err)
+		return nil, err
+	}
+
+	cm, err := impl.K8sUtil.GetConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, DevtronUniqueClientIdConfigMap, client)
+	if errStatus, ok := status.FromError(err); !ok || errStatus.Code() == codes.NotFound || errStatus.Code() == codes.Unknown {
+		// if not found, create new cm
+		cm = &v1.ConfigMap{ObjectMeta: v12.ObjectMeta{Name: DevtronUniqueClientIdConfigMap}}
+		data := map[string]string{}
+		data[DevtronUniqueClientIdConfigMapKey] = util.Generate(16) // generate unique random number
+		cm.Data = data
+		_, err = impl.K8sUtil.CreateConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, cm, client)
+		if err != nil {
+			impl.logger.Errorw("exception while getting unique client id", "error", err)
+			return nil, err
+		}
+	}
+
+	if cm == nil {
+		impl.logger.Errorw("configmap not found while getting unique client id", "cm", cm)
+		return nil, err
+	}
+	dataMap := cm.Data
+	ucid := dataMap[DevtronUniqueClientIdConfigMapKey]
+	data := &PosthogData{
```

Comment:
cache UIID, fetching data from cm for each request will put unnecessary load on api server

Reply:
implemented